docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,247 @@
+  - We haven't employed this technique before in any of our fxa services
+
+### Option D - Use a CSS pre-processor like SASS
+
+- Description
+  - [SASS](https://sass-lang.com/documentation) (which stands for 'Syntactically awesome style sheets') is an extension of CSS that enables making use of variables, nested rules, inline imports and other interesting functionalities that can make writing CSS much more powerful. in a stylesheet. It comes with 2 kinds of syntax: SCSS (Sassy CSS) and Indented (simply called 'Sass'). SCSS files use .scss extension and are fully compliant with CSS syntax, whereas the SASS files are not and hence, we will be discussing only about using .scss files as our styling solution in this section.
+- Pros
+  - Sass offers little-to-no learning curve especially if we're using SCSS syntax since it's nothing but CSS.
+  - It has large community and well documented.
+  - SASS offers modularization. With the help of `@import` directive, we can import partials (small SASS files) into other SASS files.
+  - It is compatible with all versions of CSS so any available CSS libraries can be employed with SASS.
+- Cons
+  - Using SASS will add an extra step of compilation which can potentially slow down development.
+  - Source files may be small but the generated CSS could be huge.
+
+## Pros and Cons of the Email Previewing Options
+
+It’s notoriously difficult to preview what different email states look like in FxA for Engineering, Design, and Product. Currently, we're manually previewing the templates by running `yarn write-emails` in the `fxa-auth-server` root directory which doesn't account for states unless we manually change variables and run the command again. Furthermore, even the Maildev client requires us to manually trigger the email when running FxA locally so we need a robust solution to account for these challenges faced by the team.
+
+### Option A - Continue to use the `write-emails` command and maildev client
+
+- Description
+  - Continue exclusively using the `yarn write-emails` command with [maildev](https://www.npmjs.com/package/maildev) to preview emails.
+- Pros
+  - Maildev has applications beyond previewing templates as it can help ensure mail delivery and also retrieve verification code while signing up locally.
+- Cons
+  - Previewing emails with maildev will require us to manually trigger emails while FxA is running locally.
+  - Since Maildev doesn't account for viewing various states of the emails simultaneously, it could be a bit complex to change variables and triggering the mail again and again.
+
+### Option B - Use Storybook
+
+- Description
+  - Use [Storybook](https://storybook.js.org/) to preview the email templates
+- Pros
+  - Storybook is easy to setup and configure.
+  - Storybook deployment has worked great so far for a few other FxA services so hooking up the client for `fxa-auth-server` to view the modernized components might result in consistency.
```

Comment:
Maybe this is obvious, but the main problem we're trying to solve in the "Email previewing" section is to actually have a tool to preview all of the different email states, and Storybook makes that possible. Could be worth noting as the first pro.